app/services/katello/pulp3/repository.rb

```diff
@@ -111,10 +117,26 @@ def delete_remote(href = repo.remote_href)
         remotes_api.delete(href) if href
       end
 
-      def list_remotes(args)
+      def self.delete_remote(smart_proxy, href)
+        remotes_api(smart_proxy).delete(href) if href
+      end
+
+      def remotes_list(args)
         remotes_api.list(args).results
       end
 
+      def self.delete_orphan_remotes(smart_proxy, repo_types = pulp3_enabled_repo_types(smart_proxy))
+        repo_types.each do |repo_type|
+          pulp3_class = repo_type.pulp3_service_class
+          remotes = pulp3_class.remotes_list(smart_proxy, {})
+          repos_on_capsule = ::Katello::Pulp3::Repository.new(nil, smart_proxy).list({})
+          pulp_ids = repos_on_capsule.collect { |repo| repo.name }
+          remotes.each do |remote|
+            pulp3_class.delete_remote(smart_proxy, remote['_href']) unless pulp_ids.include?(remote['name]'])
```

Comment:
i think _href has been renamed to 'pulp_href'.  Similarly to orphan distributions, this needs to return the tasks that are triggered